mantidproject/mantid
# Scale large intensities (and errors) in save reflections so doesn't overflow into previous column

**Description of work.**

Scale large intensities (and errors) in save reflections so doesn't overflow into previous column - and added tests

**To test:**

(1) Run this script to make a peak workspace with a peak with large intensity and error
```
ws = LoadEmptyInstrument(InstrumentName='SXD', OutputWorkspace='empty_SXD')
peaks = CreatePeaksWorkspace(InstrumentWorkspace=ws, NumberOfPeaks=0)
SetUB(peaks, UB=np.diag([1.0 / 3.9, 0.25, 0.1]))
pk = peaks.createPeakHKL([1,1,1])
pk.setIntensity(3.543E10)
pk.setSigmaIntensity(np.sqrt(pk.getIntensity()))
peaks.addPeak(pk)
```
(2) Call `SaveReflections` with different formats and check the intensities have been scaled (the intensity doesn't run into the L column).

Note for formats GSAS and SHELX this algorithm calls `SaveHKL` - you will see this inserts a row of 0s at the end of the file which appears to be deliberate...
https://github.com/mantidproject/mantid/blob/4892bad152edebf96bb8def2168a577e9182a1f6/Framework/Crystal/src/SaveHKL.cpp#L507-L513

*There is no associated issue.*





---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
